airbyte-scheduler/src/main/java/io/airbyte/scheduler/worker_run/TemporalWorkerRunFactory.java

```diff
@@ -122,7 +123,11 @@ private static TemporalJobType toTemporalJobType(ConfigType jobType) {
   private OutputAndStatus<JobOutput> toOutputAndStatus(CheckedSupplier<JobOutput, TemporalJobException> supplier) {
     try {
       return new OutputAndStatus<>(JobStatus.SUCCEEDED, supplier.get());
-    } catch (TemporalJobException e) {
+    } catch (TemporalJobException | TemporalException e) {
+      // while from within the temporal activity we throw TemporalJobException, Temporal wraps any
+      // exception thrown by an activity in an ApplicationFailure exception, which gets wrapped in
+      // ActivityFailure exception which get wrapped in a WorkflowFailedException. We will need to unwrap
+      // these later, but for now we just catch the parent.
```

Comment:
should we create an issue for this?

Reply:
normally i would say yes, but i plan to just handle this in this open PR https://github.com/airbytehq/airbyte/pull/2385